docs/markdown/recursor/settings.md

```diff
@@ -641,32 +658,37 @@ together with this one to prevent the stack from overflowing.
 ## `max-tcp-clients`
 * Integer
 * Default: 128
+* Available since: 2.9.18
 
 Maximum number of simultaneous incoming TCP connections allowed.
 
 ## `max-tcp-per-client`
 * Integer
 * Default: 0 (unlimited)
+* Available since: 2.9.21
 
 Maximum number of simultaneous incoming TCP connections allowed per client
 (remote IP address).
 
 ## `max-tcp-queries-per-connection`
 * Integer
 * Default: 0 (unlimited)
+* Available since: 4.0.0
```

Comment:
I didn't check the other ones, but `max-tcp-queries-per-connection` is currently not available in 4.0.0, only in master.